### ISSUE TITLE
[Accessibility][Global] [Icons] [Title] > SVG icons missing appropriate role attributes; informative icons should use role="img", decorative icons should use role="none" and aria-hidden="true".

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1945]
[A11Y-589](https://jira.cengage.com/browse/A11Y-589)

**Environment:**
Windows 11 / Chrome 137 / JAWS 2025
Windows 11 / Edge 136 / NVDA 2025
Mac 15 / Safari / VoiceOver

**Steps to Reproduce:**
1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/icon/
2. Turn on a screen reader (JAWS/NVDA/VoiceOver).
3. Launch 'Icon' component page from left navigation.
4. On the 'Icon' page, under 'Title'.
5. Navigate through the page and inspect the SVG icons using browser dev tools or accessibility tree.
4. Observe the presence or absence of role attribute on SVG elements.

**Actual Result:**
SVG icons on the page are missing the role attribute. As a result, SVG icons will not be compatible across assistive technologies and browsers, potentially causing accessibility issues for users relying on screen readers or other assistive tools.

Refer to Screencast: https://app.screencast.com/A6mtMPCRoT5eU

Instance: This applies to all the <svg> icons present in the Icon component

**Expected Result:**
Informative icons should have role="img" and a descriptive accessible name (via aria-label, title, or aria-labelledby).
Decorative icons should have role="none" and aria-hidden="true" to ensure they are ignored by assistive technologies.

**User Impact:**
Screen reader users may:
Miss important visual cues if informative icons are not announced.
Experience confusion or distraction if decorative icons are announced unnecessarily.

**WCAG Guidelines:**
1.1.1 Non-text Content (Level A)

**Bug Fix Recommendation:**
Update SVG elements based on their purpose:

**Comment by nikitaorliak-cengage** [comment]
We don't have the `Icon` component; users should add role/other attributes themselves. We can mention that in the documentation.

**Comment by silvalaura** [comment]
Approach:
- add optional properties for hidden and role in [IconProps](https://github.com/cengage/react-magma-icons/blob/main/src/IconProps.ts) 
- update the docs site to explain this accessibility rule and mention `role`
- update our docs examples
- go through all components and ensure aria-hidden and aria-role are properly set when necessary

**Comment by olehnoskov-cengage** [comment]
@silvalaura 
I have another approach, and I wonder, what do you think about it:
- we do not add new optional properties for hidden and role in [IconProps](https://github.com/cengage/react-magma-icons/blob/main/src/IconProps.ts)
- we add properties for hidden and role in [SvgIconProps](https://github.com/cengage/react-magma-icons/blob/main/src/SvgIcon.tsx#L17)
- we add default rule like this:
`ariaHidden = title ? false : true AND
  role = title ? "img" : "none",` because we have these rules:
    - consider automatically setting aria-hidden to true if no title is passed
    - Decorative icons should have `role="none"` `and aria-hidden="true"` to ensure they are ignored by assistive technologies.
    if not 
    - we go to the other option -> set have `role="img"` and a descriptive accessible name (via aria-label, title, or aria-labelledby)

In the result, we avoid updating many components, because default values are already in place.
If needed, we can add a title for icons in cases where we want to update the `role` and `aria-hidden` attributes.
It also automatically adds aria-hidden for all components:
- AiButton
- Accordion
- Banner
- Button Group
- Dropdown
- Hyperlink
- IconButton
- Pagination
- Popover
- Search
- Select
- Tabs
- Tag
- TimePicker
- Toast
- Toogle Button
- Dropzone

Example of docs

<img width="1203" height="604" alt="Image" src="https://github.com/user-attachments/assets/d5b4690f-e357-4f61-8b1a-cf2cb0a3a91d" />

**Comment by silvalaura** [comment]
@olehnoskov-cengage I like this idea! Let's move forward with it

**Comment by olehnoskov-cengage** [comment]
Should fix this bug as well.

[Nav Tabs - Need to hide icons from assistive tech](https://github.com/cengage/react-magma/issues/1239#issuecomment-3543852566)

**Comment by silvalaura** [comment]
@olehnoskov-cengage is this ready for review, or is there more work needed?

**Comment by olehnoskov-cengage** [comment]
@silvalaura 
No, it needs more work, because updating `react-magma-dom` to the latest version of `react-magma-icons` didn't change the implementation of all icons. I asked @moathabuhamad-cengage to help me with this issue, because it doesn't work with the `3.2.2` version, BUT when I link locally to the latest `react-magma-icons` version, it works well.
I wonder what the issue can be in this case?
I rechecked twice the last changes in [3.2.2](https://www.npmjs.com/package/react-magma-icons/v/3.2.2?activeTab=code#:~:text=ariaHidden%3F%3A%20boolean%3B) version, tried to clean, rebuild the app, but it didn't help.
Even, if I go to the `react-magma-docs` -> `node_modules` -> `react-magma-icons` I can see the updated code 
 but it doesn't work.

**Comment by moathabuhamad-cengage** [comment]
fixed on v5 dev 

<img width="1913" height="1089" alt="Image" src="https://github.com/user-attachments/assets/6a4626c0-956b-45f6-9c27-37b588ffa803" />

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/icon/
- V4: https://next--upbeat-sinoussi-f675aa.netlify.app/api/icon/

Everything works as expected, no bugs found.
